### PR TITLE
[3303] Use common identification strategy for transient representations

### LINKED ISSUE #3303: Use a common identification strategy for transient representations
We need to be able to identify clearly if an instance of a representation is transient or not. For that, applications contributing representations should use the same identification strategy to let us find out if a representation is transient or not.

See the ADR 137 for additional details.

### PR DESCRIPTION
**Pull request by pcdavid**

### CODE REVIEW

File: packages/forms/backend/sirius-components-collaborative-widget-reference/src/test/java/org/eclipse/sirius/components/collaborative/widget/reference/providers/ModelBrowserDescriptionProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+/**
+ * Unit tests for the model browser description provider.
+ *
+ * @author pcdavid
+ */
+public class ModelBrowserDescriptionProviderTest {
+
+    @Test
+    public void testTreeIdFormat() {
```
Comment (sbegaudeau) [inline packages/forms/backend/sirius-components-collaborative-widget-reference/src/test/java/org/eclipse/sirius/components/collaborative/widget/reference/providers/ModelBrowserDescriptionProviderTest.java]: While this test may be interesting, the real test that I'll need is, in Sirius Web, to be able to do:

```
query getRepresentationMetadata($editingContextId: ID!, $representationId: ID!) {
  viewer {
    editingContext(editingContextId: $editingContextId) {
      representation(representationId: $representationId) {
        id
        description {
          id
        }
      }
    }
  }
}
```

And see that working for both transient and persisted representations without having to be subscribed to the representation event processor. This is what's blocking me to move forward.

File: packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/UUIDParser.java [hunk 2]
```diff
@@ -26,6 +26,15 @@
...
     private final Logger logger = LoggerFactory.getLogger(UUIDParser.class);
 
+    public boolean canHandle(String id) {
+        try {
+            UUID.fromString(id);
+            return true;
+        } catch (IllegalArgumentException exception) {
+            return false;
+        }
+    }
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/UUIDParser.java]: I don't think we should be making assumptions based on UUID's only.

File: packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/representation/services/PersistentRepresentationMetadataProvider.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    private final IRepresentationDataSearchService representationDataSearchService;
+
+    public PersistentRepresentationMetadataProvider(IRepresentationDataSearchService representationDataSearchService) {
+        this.representationDataSearchService = Objects.requireNonNull(representationDataSearchService);
+    }
+
+    @Override
+    public boolean canHandle(String representationId) {
+        // Persistent representations, and only them, use valid UUIDs for their ids.
+        return new UUIDParser().canHandle(representationId);
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/representation/services/PersistentRepresentationMetadataProvider.java]: Test if the representation exists in the database with representationDataSearchService instead. "Our persistent representations and only them will use UUIDs" that may not be the case for possible extensions. They should try to respect that but by checking in the database we have a definitive answer.

File: packages/sirius-web/backend/sirius-web-services-api/src/main/java/org/eclipse/sirius/web/services/api/id/IDParser.java [hunk 4]
```diff
@@ -27,6 +27,15 @@ public class IDParser {
...
 
+    public boolean canHandle(String id) {
+        try {
+            UUID.fromString(id);
+            return true;
+        } catch (IllegalArgumentException exception) {
+            return false;
+        }
+    }
+
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-services-api/src/main/java/org/eclipse/sirius/web/services/api/id/IDParser.java]: Same comment as in the other architecture, the real test to determine if we can handle representation metadata for persisted representation is whether or not they exist.

File: packages/forms/backend/sirius-components-collaborative-widget-reference/src/test/java/org/eclipse/sirius/components/collaborative/widget/reference/providers/ModelBrowserDescriptionProviderTest.java (hunk 1 above)
Comment (sbegaudeau) [reply packages/forms/backend/sirius-components-collaborative-widget-reference/src/test/java/org/eclipse/sirius/components/collaborative/widget/reference/providers/ModelBrowserDescriptionProviderTest.java]: And thus not having to rely on the `EditingContextEventProcessorRegistry` in `EditingContextRepresentationDataFetcher`

**Review by sbegaudeau — CHANGES_REQUESTED** [review]

**Comment by pcdavid** [comment]
This is not actually ready for review, but I wanted to get the test results.

**Comment by pcdavid** [comment]
The 2 e2e test failures in the last build are unrelated (node aspect ratio), and unreliable (they were OK in the previous build on the exact same commit).

**Comment by sbegaudeau** [comment]
I've rebased the PR with some additional tests leveraging the work done in the PR (retrieval of the palette, retrieval of the reference widget options and retrieval of the help text on a form widget). I'll look at one or more additional details, review the new code coverage score which should be even better now and merge it.

File: packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/representation/services/RepresentationMetadataSearchService.java [hunk 5]
```diff
@@ -50,8 +50,8 @@ public Optional<RepresentationMetadata> findByRepresentationId(String representa
 
     private Optional<RepresentationMetadata> findTransientRepresentationById(String representationId) {
         Optional<RepresentationMetadata> representationMetadata = Optional.empty();
-        if (representationId.startsWith(ExplorerDescriptionProvider.REPRESENTATION_ID)) {
-            representationMetadata = Optional.of(new RepresentationMetadata(ExplorerDescriptionProvider.REPRESENTATION_ID, Tree.KIND, ExplorerDescriptionProvider.REPRESENTATION_NAME, ExplorerDescriptionProvider.DESCRIPTION_ID));
+        if (representationId.startsWith(ExplorerDescriptionProvider.PREFIX)) {
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/representation/services/RepresentationMetadataSearchService.java:56]: This is a duplicate of RepresentationApplicationService1findRepresentationMetadataById, it was a mistake on my side.

**Review by sbegaudeau — APPROVED** [review]

**Pull request merged**